### PR TITLE
limit image size tool

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

File: tools/coco_limit_image_size.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    with open(args.annotation) as f:
+        content = json.load(f)
+
+    resized = 0
+
+    for i, image_info in enumerate(tqdm(content['images'])):
+        path = os.path.join(args.input_root, image_info['file_name'])
+        image = cv2.imread(path)
+        if image is None:
+            print(path)
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py]: Could you please make this message more informative?

File: tools/coco_limit_image_size.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            new_size = int(floor(image.shape[1] / ratio)), int(floor(image.shape[0] / ratio))
+            image = cv2.resize(image, new_size)
+            resized += 1
+
+        if i % 1000 == 0:
+            print(f'resized {resized} of {i + 1}')
+
+        output_path = os.path.join(args.output_root, image_info['file_name'])
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
+        cv2.imwrite(output_path, image)
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py:61]: Wouldn't it be faster to get the size of the image as you did in other tools and read/resize/write only those that have to be processed according to that size?

File: tools/coco_limit_image_size.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        height_ratio = image.shape[0] / args.limit_image_size[0]
+        width_ratio = image.shape[1] / args.limit_image_size[1]
+        if height_ratio > 1 or width_ratio > 1:
+            ratio = height_ratio if height_ratio > width_ratio else width_ratio
+            new_size = int(floor(image.shape[1] / ratio)), int(floor(image.shape[0] / ratio))
+            image = cv2.resize(image, new_size)
+            resized += 1
+
+        if i % 1000 == 0:
+            print(f'resized {resized} of {i + 1}')
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py]: Is it a useful message? I'm a bit concerned that it breaks the progressbar and wondering wouldn't it make sense to print this statistics only after the whole dataset being processed?

File: tools/coco_limit_image_size.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        output_path = os.path.join(args.output_root, image_info['file_name'])
+        os.makedirs(os.path.dirname(output_path), exist_ok=True)
```
Comment (druzhkov-paul) [inline tools/coco_limit_image_size.py:60]: Do you need to do that on every iteration?

File: tools/coco_limit_image_size.py (hunk 2 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py:61]: Anyway image should be copied to output dir even if it less than specified size

File: tools/coco_limit_image_size.py (hunk 4 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py:60]: folder with images might have different structure, so for general case it is required.
i don't think that it takes too much time relatively to image reading/resizing/writing

File: tools/coco_limit_image_size.py (hunk 3 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py]: Fixed

File: tools/coco_limit_image_size.py (hunk 1 above)
Comment (Ilya-Krylov) [reply tools/coco_limit_image_size.py]: fixed

File: tools/coco_limit_image_size.py (hunk 2 above)
Comment (druzhkov-paul) [reply tools/coco_limit_image_size.py:61]: Agree, but it's not encoded and decoded again while copying. Though, I don't insist.

File: tools/coco_limit_image_size.py (hunk 4 above)
Comment (druzhkov-paul) [reply tools/coco_limit_image_size.py:60]: I see, Ok then.

**Review by druzhkov-paul — APPROVED** [review]

**Pull request merged**